skdownscale/pointwise_models/quantile.py

```diff
@@ -0,0 +1,363 @@
+                # calculate the data value pp[-1]
+                vals[-1] = model.predict(pp[-1].reshape(-1, 1))
+
+        return Cdf(pp, vals)
+
+    def _more_tags(self):
+        return {
+            '_xfail_checks': {
+                'check_estimators_dtypes': 'QuantileMappingReressor only suppers 1 feature',
+                'check_fit_score_takes_y': 'QuantileMappingReressor only suppers 1 feature',
+                'check_estimators_fit_returns_self': 'QuantileMappingReressor only suppers 1 feature',
+                'check_estimators_fit_returns_self(readonly_memmap=True)': 'QuantileMappingReressor only suppers 1 feature',
+                'check_dtype_object': 'QuantileMappingReressor only suppers 1 feature',
+                'check_pipeline_consistency': 'QuantileMappingReressor only suppers 1 feature',
+                'check_estimators_nan_inf': 'QuantileMappingReressor only suppers 1 feature',
+                'check_estimators_overwrite_params': 'QuantileMappingReressor only suppers 1 feature',
+                'check_estimators_pickle': 'QuantileMappingReressor only suppers 1 feature',
+                'check_fit2d_predict1d': 'QuantileMappingReressor only suppers 1 feature',
+                'check_methods_subset_invariance': 'QuantileMappingReressor only suppers 1 feature',
+                'check_fit2d_1sample': 'QuantileMappingReressor only suppers 1 feature',
+                'check_dict_unchanged': 'QuantileMappingReressor only suppers 1 feature',
+                'check_dont_overwrite_parameters': 'QuantileMappingReressor only suppers 1 feature',
+                'check_fit_idempotent': 'QuantileMappingReressor only suppers 1 feature',
+                'check_n_features_in': 'QuantileMappingReressor only suppers 1 feature',
+                'check_estimators_empty_data_messages': 'skip due to odd sklearn string matching in unit test',
+                'check_regressors_train': 'QuantileMappingReressor only suppers 1 feature',
+                'check_regressors_train(readonly_memmap=True)': 'QuantileMappingReressor only suppers 1 feature',
+                'check_regressors_train(readonly_memmap=True,X_dtype=float32)': 'QuantileMappingReressor only suppers 1 feature',
+                'check_regressor_data_not_an_array': 'QuantileMappingReressor only suppers 1 feature',
+                'check_regressors_no_decision_function': 'QuantileMappingReressor only suppers 1 feature',
+                'check_supervised_y_2d': 'QuantileMappingReressor only suppers 1 feature',
+                'check_regressors_int': 'QuantileMappingReressor only suppers 1 feature',
+                'check_methods_sample_order_invariance': 'QuantileMappingReressor only suppers 1 feature',
+            },
+        }
+
+
+class TrendAwareQuantileMappingRegressor(RegressorMixin, BaseEstimator):
+    def __init__(self, qm_estimator=None):
+        self.qm_estimator = qm_estimator
+
+    def fit(self, X, y):
```

Comment:
I'll do the same thing here (offer up some explanatory comments to add to the code but also a test of my understanding)
'''
Detrend the two input datasets then create the quantile mapping between the two detrended datasets.
'''